### PR TITLE
Consolidate fetch init type annotation.

### PR DESCRIPTION
**Pull request by alabiaga**
- delete FetchInitJsonDef and consolidate with FetchInitDef

### CODE REVIEW

File: build-system/amp.extern.js [hunk 1]
```diff
@@ -17,26 +17,32 @@
...
+ * __amp_source_origin should not be appended to the URL to allow for
+ * potential caching or response across pages.
+ *
+ * See https://developer.mozilla.org/en-US/docs/Web/API/GlobalFetch/fetch
+ *
  * Externed for use in the amp-form component when reconstructing
  * the request for SSR and then passed to runtime.
  * @typedef {{
- *   body: (!JsonObject|!FormData|undefined),
+ *   body: (!JsonObject|!FormData|!Object|!Array|undefined|string),
```
Comment (jridgewell) [inline build-system/amp.extern.js]: Revert this change.

**Comment by cathyxz** [comment]
Sorry, didn't mean to accidentally refer to this PR (that was a typo).

File: build-system/amp.extern.js (hunk 1 above)
Comment (alabiaga) [reply build-system/amp.extern.js]: Doing so breaks check-types run, though Array, I was able to remove. Certain extensions refer to the 'body' field expecting it to be of type object type or string.

File: build-system/amp.extern.js [hunk 2]
```diff
@@ -17,26 +17,32 @@
...
+ * __amp_source_origin should not be appended to the URL to allow for
+ * potential caching or response across pages.
+ *
+ * See https://developer.mozilla.org/en-US/docs/Web/API/GlobalFetch/fetch
+ *
  * Externed for use in the amp-form component when reconstructing
  * the request for SSR and then passed to runtime.
  * @typedef {{
- *   body: (!JsonObject|!FormData|undefined),
+ *   body: (!JsonObject|!FormData|!Object|undefined|string),
```
Comment (jridgewell) [inline build-system/amp.extern.js]: `string` is acceptable, but we can't accept an `Object` here. Anything that's trying to pass an `Object` should be using a `JsonObject` instead.

Comment (alabiaga) [reply build-system/amp.extern.js]: ah gotcha. Thanks!
